mds/entrypoint.sh

```diff
@@ -34,6 +47,27 @@ if [ ! -e /var/lib/ceph/mds/ceph-$MDS_NAME/keyring ]; then
       # Generate the new MDS key
       ceph auth get-or-create mds.$MDS_NAME mds 'allow' osd 'allow *' mon 'allow profile mds' > /var/lib/ceph/mds/ceph-${MDS_NAME}/keyring
    fi
+
+fi
+
+# Create the Ceph filesystem, if necessary
+if [ $CEPHFS_CREATE -gt 0 ]; then
+   FS_EXISTS=$(ceph fs ls | grep -c name:.${CEPHFS_NAME},)
+   if [ $FS_EXISTS -eq 0 ]; then
+      # Make sure the specified data pool exists
+      ceph osd pool stats ${CEPHFS_DATA_POOL} >/dev/null
+      if [ $? -ne 0 ]; then
+         ceph osd pool create ${CEPHFS_DATA_POOL} ${CEPHFS_DATA_POOL_PG}
+      fi
+
+      # Make sure the specified metadata pool exists
```

Comment:
Also, the order is intentional, here.  I don't want to create the specified pools if the filesystem already exists; they would just be wasted.  Hence, I'm _only_ creating the pools if the filesystem doesn't already exist.  It's not intuitive, but I think it's safer than creating arbitrary pools just because a user told the mds container to create the filesystem without reading the docs on it.